UWP (Win10) support

niklaswolf opened the pull request:
Hi @endel, as I promised in issue #8, I adapted the Client to also support UWP (Windows 10). It was a bit tricky, but I managed to get it working.
I have to mention, that this is only tested in Unity directly and on Microsoft Hololens, which is running UWP apps (Win10). But because UWP is the same set of APIs for other devices, it should work for them too.

Some important changes:
- I had to get rid of WebSocketSharp, so I wrapped it in preprocessor-directives to not include it when the app is build with an UWP-app as target, hope this is okay.
- Added another section to WebSocket.cs which uses the UWP-APIs for connection etc.
- had to add another MessagePack.dll for UWP, named it MessagePackUWP. I then edited the import-Settings in Unity, so that the right dll gets included for the different target-platforms.
- added some public fields to ColyseusClient.cs, so that you can set servername, port and roomname directly from Unity

I don't know, if you want to merge this into the General Unity-client, or if it should stand alone from now on...
I also will have to write a little guide for building this for UWP, as I got some Errors which can be fixed by following a specific build process.

So what do you think? :)

endel commented on Assets/Colyseus/Room.cs:
```diff
@@ -76,9 +79,13 @@ public int id
+                this.OnJoin.Emit (this, EventArgs.Empty);
+#else
+                this.OnJoin.Invoke(this, EventArgs.Empty);
+#endif
```
Looks like `Emit` is an alias to `Invoke`. I didn't know that when I first write this. We can use always `Invoke`. Can we get rid of the conditional compilation?

endel commented on Assets/Colyseus/Room.cs:
```diff
@@ -87,24 +94,34 @@ public void SetState( MessagePackObject state, int remoteCurrentTime, int remote
 			// TODO:
 			// Create a "clock" for remoteCurrentTime / remoteElapsedTime to match the JavaScript API.
-
+               
```
No white spaces, please :)

endel commented on Assets/Colyseus/Room.cs:
```diff
@@ -87,24 +94,34 @@ public void SetState( MessagePackObject state, int remoteCurrentTime, int remote
+            this.OnUpdate.Emit(this, new RoomUpdateEventArgs(this, state, null));
+#else
+            this.OnUpdate.Invoke(this, new RoomUpdateEventArgs(this, state, null));
+#endif
```
Use same indentation

endel commented on Assets/Colyseus/Client.cs:
```diff
@@ -86,22 +88,33 @@ public void Recv()
+#if !WINDOWS_UWP
+            Console.WriteLine(code);
+#else
+            Debug.Log("Message Code: "+code);
```
We can remove this log completely. It shouldn't be there in the first place :)

endel commented on Assets/Colyseus/Client.cs:
```diff
@@ -86,22 +88,33 @@ public void Recv()
-			// Parse roomId or roomName
-			Room room = null;
+            // Parse roomId or roomName
+            Room room = null;
```
indentation

endel commented on Assets/ColyseusClient.cs:
```diff
@@ -7,15 +7,19 @@
+    Room chatRoom;
+    public string serverName = "localhost";
+    public string port = "3553";
+    public string roomName = "chat";
```
Please also review indentation of this whole file.

endel commented:
Hey @niklaswolf, did you had a chance to have look at this?

niklaswolf commented:
Hi @endel, yep I have read your comments, will have some time implementing it this weekend ;)
Sorry for not answering, but I was a bit busy ;)

endel commented:
No worries! Great, thanks!

niklaswolf commented:
Hi @endel, I have changed the things you requested. Maybe the indentation is still bad, but I don't understand why Visual Studio is showing different (correct) indentation than other editors. I tried my best ;)

I also added a little section to the Readme with build instructions for UWP. Additionally I fixed the Readme markup for the c#-code, so now it should show up properly again ;)

endel commented:
Awesome, thanks for reviewing. I've invited you to the organization, feel free to tidy things up here and send more pull-requests if you want to :)
